docs/architecture/grpc-for-wcf-developers/introduction.md

```diff
@@ -1,11 +1,61 @@
+
+## Microservices
+
+In microservice architectures, large applications are built as a collection of smaller modular services. Each component performs a  specific task or process, and components are designed to work inter-operably but can be isolated as necessary.
+
+Advantages to microservices include:
+
+- Changes and upgrades can be handled independently.
+- Error handling becomes more efficient as problems can be traced to specific services which are then isolated, rebuilt, tested and redeployed independent of the other services.
+- Scalability can be confined to the specific instances or services not the whole application.
+- Development can happen across teams.
+
+The move towards increasing virtualization, cloud computing and the Internet of Things have contributed to the ongoing rise of microservices. However, microservices are was not without their challenges. The fragmented/decentralized infrastructure placed more emphasis on the need for simplicity and speed when communicating between services. This in turn drew attention to the sometimes laborious and contorted nature of SOAP.
+
+It was into this environment that gRPC was launched, 10 years after Microsoft first released WCF. Evolved directly from Google’s internal infrastructure RPC (Stubby), gRPC was never based on the same standards and protocols that had informed the parameters of many earlier RPCs. Additionally, gRPC was only ever based on HTTP/2 and was therefore able to draw on the new capabilities that advanced transport protocol provided; in particular, bi-directional streaming, binary messaging and multiplexing.
+
+## About this guide
+
+This guide covers the key features of gRPC.  The early chapters take a high level look at the main features of WCF and compares them to gRPC. It identifies where the are direct correlations between WCF and gRPC and also where gRPC offers an advantage. Where there is no correlation between WCF and gRPC or where gRPC is not able to offer an equivalent solution to WCF, this guide will suggest workarounds or places to go for further information.
+
+Using a set of sample WCF applications, Chapter 5 is a deep dive look at a converting the main types of WCF service (simple request-reply, one-way, and streaming) to their equivalents in gRPC.
+
+The final section of the book looks at how to get the best from gRPC in practice, including using additional tools like Docker containers or Kubernetes to leverage to the efficiency of gRPC, and a detailed look at the monitoring with logging, metrics and distributed tracing.
+
+## Whom this guide is for
+
+This guide was written for developers working in .NET or .NET Core who have previously used WCF and who are seeking to migrate their applications to a modern RPC environment for .NET Core 3.0 and later versions. It is hoped that the guide will also be of use more generally for developers upgrading or considering upgrading to .NET Core 3.0 who want to use the built in gRPC tool.
```

Comment:
.NET or .NET Core -> did you mean .NET Framework?

"It is hoped" sounds strange